overlord/ifacestate/ifacemgr.go

```diff
@@ -286,6 +287,50 @@ func (m *InterfaceManager) ConnectionStates() (connStateByRef map[string]Connect
 	return ConnectionStates(m.state)
 }
 
+// ResolveDisconnect resolves potentially missing plug or slot names and returns
+// a fully populated connection reference. It resorts to repository's
+// ResolveDisconnect method for normal disconnect, but uses connection from the
+// state if forget flag is true, so it is able to resolve connections for
+// non-existing plugs or slots.
+// The state must be locked by the caller.
+func (m *InterfaceManager) ResolveDisconnect(plugSnapName, plugName, slotSnapName, slotName string, forget bool) ([]*interfaces.ConnRef, error) {
+	if !forget {
+		return m.repo.ResolveDisconnect(plugSnapName, plugName, slotSnapName, slotName)
+	}
+
+	// for convienience, empty plug snap means system snap
+	if plugSnapName == "" {
+		plugSnapName = mapper.SystemSnapName()
+	}
+	if plugName == "" {
+		return nil, fmt.Errorf("invalid empty plug name")
+	}
+	if slotName == "" {
+		return nil, fmt.Errorf("invalid empty slot name")
+	}
+	// for convienience, empty slot snap means system snap
+	if slotSnapName == "" {
```

Comment:
the other of the ifs is a bit odd,  I would expact plugSnapName, plugName, slotSnapName, slotName